net-vpn/i2p/i2p-2.1.0.ebuild

```diff
@@ -0,0 +1,199 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="test"
+
+inherit java-pkg-2 java-ant-2 systemd
+
+DESCRIPTION="A privacy-centric, anonymous network"
+HOMEPAGE="https://geti2p.net"
+SRC_URI="https://files.i2p-projekt.de/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+LANGS=(
+	ar az bg ca cs da de el en es es-AR et fa fi fr gl he hi hr hu id it ja ko ku mg nb nl nn pl pt pt-BR ro ru sk sl sq
+	sr sv tk tr uk vi zh zh-TW
+)
+IUSE="${LANGS[@]/#/l10n_}"
+
+CP_DEPEND="
+	dev-java/java-getopt:1
+	dev-java/java-service-wrapper:0
+	dev-java/jstl-api:0
+	dev-java/jstl:0
+	net-libs/nativebiginteger:0
+	sys-devel/gettext:0[java]
+	>=www-servers/tomcat-9.0.73:9
+"
+DEPEND="
+	${CP_DEPEND}
+	>=virtual/jdk-1.8:*
+	test? (
+		dev-java/ant-junit4:0
+		dev-java/hamcrest:0
+		dev-java/hamcrest-core:1.3
+		dev-java/junit:4
+		dev-java/mockito:4
+	)
+"
+RDEPEND="
+	${CP_DEPEND}
+	acct-user/i2p
+	acct-group/i2p
+	>=virtual/jre-1.8:*
+"
+
+EANT_BUILD_TARGET="preppkg-base"
+# no scala as depending on antlib.xml not installed by dev-lang/scala
+EANT_TEST_TARGET="junit.test"
+JAVA_ANT_ENCODING="UTF-8"
+
+DOCS=(README.md history.txt)
+
+PATCHES=(
+	"${FILESDIR}/${PV}-fix-getopt-classpath.patch"
+)
+
+JAVA_PKG_NO_CLEAN=(
```

Comment:
The `JAVA_PKG_NO_CLEAN` array would work only starting from EAPI 9, see [this commit](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=7d1708a45b0471753da0a59442e5a1413c335ba0).
Before EAPI 9 you'd better put this line in src_prepare()
```
java-pkg_clean ! -path "./apps/jetty/jetty-distribution-*"
```